components/gfx/text/text_run.rs

```diff
@@ -304,6 +304,24 @@ impl<'a> TextRun {
         })
     }
 
+    /// Returns the index in the range of the first glyph advancing over given advance
+    pub fn range_index_of_advance(&self, range: &Range<ByteIndex>, advance: Au) -> Option<usize> {
+        if range.is_empty() {
```

Comment:
Is this check necessary? Doesn't `natural_words_in_range` handle empty ranges?

Also, this function seems to always return `Some(..)`, so why not returning just an `usize`?